greentest/test__issues461_471.py

```diff
@@ -0,0 +1,37 @@
+'''Test for GitHub issues 461 and 471.
+
+When moving to Python 3, handling of KeyboardInterrupt exceptions caused
+by a Ctrl-C raise an exception while printing the traceback for a
+greenlet preventing the process from exiting. This test tests for proper
+handling of KeyboardInterrupt.
+'''
+
+import sys
+
+if sys.argv[1:] == ['subprocess']:
+    import gevent
+    def task():
+        sys.stdout.write('ready\n')
+        sys.stdout.flush()
+        gevent.sleep(30)
+    try:
+        gevent.spawn(task).get()
+    except KeyboardInterrupt:
+        pass
+else:
+    import signal
+    from subprocess import Popen, PIPE, TimeoutExpired
+    if sys.platform.startswith('win'):
+        from subprocess import CREATE_NEW_PROCESS_GROUP
+        kwargs = {'creationflags': CREATE_NEW_PROCESS_GROUP}
+    else:
+        kwargs = {}
+    p = Popen([sys.executable, __file__, 'subprocess'], stdout=PIPE, **kwargs)
+    p.stdout.readline()
+    p.send_signal(signal.SIGINT)
+    try:
+        p.wait(3)
+    except TimeoutExpired:
+        p.terminate()
+        sys.exit(1)
+    sys.exit(p.returncode)
```

Comment:
Just a couple of idea.s

For making this Py2/3-compatible, depending on when in the test suite this test is about to run (after Timeout tests it should be!), it might be a good idea to just use `gevent.Timeout` with a polling-based approach as a *simple* cross-platform timeout solution, like so (untested):

```python
rc = None        
with gevent.Timeout(3, False):
    while rc is None:
        gevent.sleep(0.01)
        rc = p.poll()
    sys.exit(rc)
p.terminate()     
sys.exit(1)
```

You might also want to do `assert p.stdout.readline() == "ready"`, when you're readline()ing anyway. I am not 100 % sure right now, but if there is a chance for a race condition among poll() and the Timeout exception, the argument to the last `sys.exit()` should not simply be 1, but sth like `rc if rc is not None else 1`. However, I think that should never be of relevance, considering the large timeout value of three seconds.